docs/usage.md

```diff
@@ -275,27 +274,100 @@ You can also supply a run name to resume a specific run: `-resume [run-name]`. U
+
+As mentioned at the beginning of this section it may also be necessary for users to overwrite the options passed to modules to be able to customise specific aspects of the way in which a particular tool is executed by the pipeline. Given that all of the default module options are stored in the pipeline's `modules.config` as a [`params` variable](https://github.com/nf-core/rnaseq/blob/4c27ef5610c87db00c3c5a3eed10b1d161abf575/conf/modules.config#L24-L25) it is also possible to overwrite any of these options via a custom config file.
+
+Say for example we want to append an additional, non-mandatory parameter (i.e. `--outFilterMismatchNmax 16`) to the arguments passed to the `STAR_ALIGN` module. Firstly, we need to copy across the default `args` specified in the [`modules.config`](https://github.com/nf-core/rnaseq/blob/4c27ef5610c87db00c3c5a3eed10b1d161abf575/conf/modules.config#L71) and create a custom config file that is a composite of the default `args` as well as the additional options you would like to provide. This is very important because Nextflow will overwrite the default value of `args` that you provide via the custom config.
+
+As you will see in the example below, we have:
+
+* appended `--outFilterMismatchNmax 16` to the default `args` used by the module.
+* changed the default `publish_dir` value to where the files will eventually be published in the main results directory.
+* appended `'bam':''` to the default value of `publish_files` so that the BAM files generated by the process will also be saved in the top-level results directory for the module. Note: `'out':'log'` means any file/directory ending in `out` will now be saved in a separate directory called `my_star_directory/log/`.
+
+```nextflow
+params {
+    modules {
+        'star_align' {
+            args          = "--quantMode TranscriptomeSAM --twopassMode Basic --outSAMtype BAM Unsorted --readFilesCommand zcat --runRNGseed 0 --outFilterMultimapNmax 20 --alignSJDBoverhangMin 1 --outSAMattributes NH HI AS NM MD --quantTranscriptomeBan Singleend --outFilterMismatchNmax 16"
+            publish_dir   = "my_star_directory"
+            publish_files = ['out':'log', 'tab':'log', 'bam':'']
+        }
+    }
```

Comment:
There are a lot of lines here about staralign - I worry a little they detract from the main usage. Could/Should these be made more specific or moved to a different md file?

Reply:
So the idea behind adding these particular docs and using `STAR_ALIGN` as an example is that it forms a general narrative to users as to how to change the resources for DSL2 pipelines and to amend the module options. I needed to find a process that we could use throughout to be coherent with the explanations and to be able to point to real code in the rnaseq pipeline that would help users troubleshoot and figure this sort of thing out for themselves - `STAR_ALIGN` is one of those processes that has to be tweaked quite often which is why I chose that as an example. Took me the best part of the day to write these two paragraphs! But they have been pointed to numerous times already where users have wanted to tweak such options. I digress, you are absolutely right about moving the docs - we have discussed this a few times but we don't have an official home for them yet, at which point they will be moved out of the DSL2 pipeline template 👍🏽